lib/router.js

```diff
@@ -34,4 +41,65 @@ const routes = [
   { path: '*', component: NotFound }
 ];
 
-export default new VueRouter({ routes });
+
+
+////////////////////////////////////////////////////////////////////////////////
+// INITIALIZE
+
+Vue.use(VueRouter);
+const router = new VueRouter({ routes });
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// HISTORY
+
+let historySize = 0;
+
+router.beforeEach((to, from, next) => {
+  historySize += 1;
+  next();
+});
+
+// Goes back to the page last visited in the app, navigating to `fallback` if
+// the app was just loaded and is on its first page.
+const safeBack = (fallback) => {
```

Comment:
can i suggest naming this `backOrElse`? "safe" is a bit ambiguous.

Reply:
and actually, i don't think we want to do this operation this way at all. can we put the return-to URL in the queryparam? i think it is unexpected that logging in pushes your browser history backward.